### PR TITLE
use current version during first time deployment

### LINKED ISSUE #955: delete and redeploy of same app sometimes uses old revision
i tried deleting (from the UI) and redeploying my app on Fermyon cloud and sometimes it seems to be serving old revision (until I deploy the app 2nd time).

### PR DESCRIPTION
**Pull request by rajatjindal**
fixes #955 

Signed-off-by: Rajat Jindal <rajatjindal83@gmail.com>

### CODE REVIEW

**Comment by michelleN** [comment]
@rajatjindal Thanks for this. I see there is a `[WIP]` in the title. Is this still a WIP or is it ready for review?

**Comment by michelleN** [comment]
@rajatjindal Thanks for this. I see there is a [WIP] in the title. Is this still a WIP or is it ready for review?

**Comment by rajatjindal** [comment]
Hi @michelleN , i just edited out the `WIP` tag, and this PR should be ready for review. IMO we still need to fix #971 for this fix to be complete though

File: src/commands/deploy.rs [hunk 1]
```diff
@@ -384,17 +384,21 @@ impl DeployCommand {
                 existing_channel_id
             }
             Err(_) => {
-                let range_rule = Some(bindle_id.version_string());
                 let app_id = CloudClient::add_app(&client, &name, &name)
                     .await
                     .context("Unable to create app")?;
+
+                let active_revision_id = self
```
Comment (itowlson) [inline src/commands/deploy.rs:391]: We don't seem to be calling `CloudClient::add_revision` on this path (as we do on the "app exists" path.  Do we understand why it is unnecessary on this path but necessary on the other?  Could we maybe add a comment for the benefit of a maintainer who comes across this and is puzzled like me?  Thanks!

**Review by itowlson — CHANGES_REQUESTED** [review]
This looks good but I had one question - and even if my question is a no-op I think it would benefit from a clarifying comment - thanks!

**Comment by rajatjindal** [comment]
Hi Ivan,

Thanks for your question, it totally makes sense to add it to doc. Will do it in morning.

Recording here as well

What i noticed is that during create app, "import revisions" event is triggered which reconciles bindle revisions in our storage for the given user, so add revision is not required here.

I think one more testcase i should verify explicitly is creation of 2nd app, however because i tested with Fermyon cloud, that is probably covered already. 

I will update here on how it goes

File: src/commands/deploy.rs (hunk 1 above)
Comment (rajatjindal) [reply src/commands/deploy.rs:394]: https://github.com/deislabs/hippo/blob/e83bda6858c5d5b8db234f6f07bbf4b1afb5f8d2/src/Application/Apps/EventHandlers/InitialRevisionImport.cs#L9  - on app creation, we import all existing revisions into db. therefore we don't need add_revision to be called explicitly (I did added that initially, but it was creating a duplicate revision entry into db).

added this as a comment in code as well. 

thank you for pointing it out

**Pull request merged**